scripts/gatsby-source-iiif/src/getCenturyTagsFromDate/index.js

```diff
@@ -3,32 +3,46 @@ module.exports = (dates) => {
   if (!dates) {
     return ['undated']
   }
-  console.log(dates)
-  if (Array.isArray(dates)) {
-    console.log(dates)
-    dates = dates[0]
-  }
-  dates = dates.toString()
-  console.log(dates)
-  const mappedDates = dates.match(/([0-9]{4})/g)
-  if (!mappedDates) {
-    console.error('date not mapped', dates)
-    return ['undated']
+  // ensure it is an array
+  if (!Array.isArray(dates)) {
+    dates = [dates]
   }
+  dates = dates.map((d) => {
+    d = d.toString()
+    d = fixLibraryDates(d)
+    // find all the years 3 and 4 digit in here.
+    // still a problem with 2 digit ones.
+    d = d.match(/([0-9]{3,4})/g)
+    return d
+  }).filter((d) => d)
 
-  let years = dates.match(/([0-9]{4})/g).map((year) => {
-    year = Math.ceil(year / 100)
-    return getNumberWithOrdinal(year) + ' Century'
+  let tags = dates.map((d) => {
+    return d.map((year) => {
+      year = parseInt(year)
```

Comment:
Use `year = parseInt(year, 10)`

Reply:
This should probably also have a default value assigned if the year parseInt fails maybe, unless null is acceptable. Without the radix, something like "1920a" would get parsed as "102922", the decimal value of an inferred hex type.

Reply:
I don't think at this point it is very likely to fail.  as it is only acting on the years the regular expression is parsing out above.  There I am removing all rows that don't have matches.  
But it is a good point.  I think I will leave it as a hard fail at the moment to indicate clearly there is a case that we need to trap.  Since I have a bunch more cases to look at after  decisions are made in the metadata team.